Extension/package.json

```diff
@@ -5092,27 +5092,27 @@
       "editor/context": [
         {
           "command": "C_Cpp.SwitchHeaderSource",
-          "when": "config.C_Cpp.intelliSenseEngine != 'Disabled' && editorLangId == 'c' || config.C_Cpp.intelliSenseEngine != 'Disabled' && editorLangId == 'cpp' || config.C_Cpp.intelliSenseEngine != 'Disabled' && editorLangId == 'cuda-cpp'",
+          "when": "(config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled') && editorLangId == 'c' || (config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled') && editorLangId == 'cpp' || (config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled') && editorLangId == 'cuda-cpp'",
```

Comment:
I don't think this condition is correct.  The part in parenthesis will always evaluate to `true`.  I think you need to use `&&` instead of `||` for the `default` vs `Default` check.